internal/future/statestore/redis.go

```diff
@@ -94,29 +96,208 @@ func redisCloser(closer func() error) {
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"error": err.Error(),
+		}).Error("failed to connect to redis")
+		return err
+	}
+	defer redisConn.Close()
+
+	redisConn.Send("MULTI")
+	key := ticket.Id
+	value, err := proto.Marshal(ticket)
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"key":   key,
+			"error": err.Error(),
+		}).Error("failed to marshal the ticket proto")
+		return err
+	}
+
+	err = redisConn.Send("SET", key, value)
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"key":   key,
+			"cmd":   "SET",
+			"error": err.Error(),
+		}).Error("failed to set the value for ticket in state storage")
+		return err
+	}
+
+	if ttl > 0 {
+		err = redisConn.Send("EXPIRE", key, ttl)
+		if err != nil {
+			redisLogger.WithFields(logrus.Fields{
+				"key":   key,
+				"cmd":   "EXPIRE",
+				"error": err.Error(),
+			}).Error("failed to set ticket expiration in state storage")
+			return err
+		}
+	}
+
+	_, err = redisConn.Do("EXEC")
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"key":   key,
+			"cmd":   "EXEC",
+			"error": err.Error(),
+		}).Error("failed to create ticket in state storage")
+		return err
+	}
+
+	return nil
 }
 
 // GetTicket gets the Ticket with the specified id from state storage. This method fails if the Ticket does not exist.
-func (rb *redisBackend) GetTicket(context.Context, string) (*pb.Ticket, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+func (rb *redisBackend) GetTicket(ctx context.Context, id string) (*pb.Ticket, error) {
+	redisConn, err := rb.redisPool.GetContext(ctx)
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"error": err.Error(),
+		}).Error("failed to connect to redis")
+		return nil, err
+	}
+	defer redisConn.Close()
+
+	value, err := redis.Bytes(redisConn.Do("GET", id))
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"key":   id,
+			"cmd":   "GET",
+			"error": err.Error(),
+		}).Error("failed to get the ticket from state storage")
+		return nil, err
+	}
+
+	if value == nil {
+		msg := fmt.Sprintf("Ticket id:%s not found", id)
+		redisLogger.WithFields(logrus.Fields{
+			"key": id,
+			"cmd": "GET",
+		}).Error(msg)
+		return nil, status.Error(codes.NotFound, msg)
+	}
+
+	ticket := &pb.Ticket{}
+	err = proto.Unmarshal(value, ticket)
+	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
+			"key":   id,
+			"error": err.Error(),
+		}).Error("failed to unmarshal the ticket proto")
+		return nil, err
+	}
+
+	return ticket, nil
 }
 
-// DeleteTicket removes the Ticket with the specified id from state storage. This method succeeds if the Ticket does not exist.
-func (rb *redisBackend) DeleteTicket(context.Context, string) error {
-	return status.Error(codes.Unimplemented, "not implemented")
+// DeleteTicket removes the Ticket with the specified id from state storage.
+func (rb *redisBackend) DeleteTicket(ctx context.Context, id string) error {
```

Comment:
What would be the behavior if receiving a request to delete a non-existing ticket?

Reply:
Same - delete succeeds if ticket does not exist. I ve added tests to actually confirm the same.